frontend/src/lib/lemon-ui/LemonCollapse/LemonCollapse.tsx

```diff
@@ -52,12 +52,14 @@ export function LemonCollapse<K extends React.Key>({
     let isPanelExpanded: (key: K) => boolean
     let onPanelChange: (key: K, isExpanded: boolean) => void
     if (props.multiple) {
+        const defaultActiveKeys = props.defaultActiveKeys ?? []
+        const defaultActiveKeysString = defaultActiveKeys.join(',')
         // eslint-disable-next-line react-hooks/rules-of-hooks
-        const [localActiveKeys, setLocalActiveKeys] = useState<Set<K>>(new Set(props.defaultActiveKeys ?? []))
+        const [localActiveKeys, setLocalActiveKeys] = useState<Set<K>>(new Set(defaultActiveKeys))
         // eslint-disable-next-line react-hooks/rules-of-hooks
         useEffect(() => {
-            setLocalActiveKeys(new Set(props.defaultActiveKeys ?? []))
-        }, [props.defaultActiveKeys])
+            setLocalActiveKeys(new Set(defaultActiveKeys))
+        }, [defaultActiveKeysString])
```

Comment:
Side effect fix - collapses would revert whenever it was rendered as this was the array object that was usually referential